Time invariance for sampling of density matrix

ec147 opened the pull request:
- Use of time invariance for more efficient sampling of the density matrix. This can be activated with the parameter `time_invariance=True`.
- At the end of the run, when checking that the trace of the density matrix is close to 1, only the warning is printed now. The error threshold has been deleted, 
  since it is overly aggressive at low T for instance.
- A test has been added.

ec147 committed:
Use of time invariance for sampling of the density matrix

Add files via upload

Add files via upload

Add files via upload

Add files via upload

Add files via upload

Add files via upload

Delete solver_core.hpp

Delete configuration.hpp

Delete solver_core.cpp

Delete qmc_data.hpp

Delete parameters.hpp

Delete impurity_trace.hpp

Add files via upload

Update CMakeLists.txt

fixes.cpp

Update qmc_data.hpp

Update solver_core.hpp

Update configuration.hpp

Update solver_core.cpp

Update parameters.hpp

Update solver_core.cpp

Update insert.hpp

Update insert.cpp

Update remove.hpp

Update remove.cpp

Update insert.hpp

Update insert.cpp

Update remove.hpp

Update remove.cpp

Update insert.cpp

Update remove.cpp

Update insert.cpp

Update remove.cpp

Update solver_core.cpp

Update update_time.hpp

Add files via upload

Delete c++/triqs_cthyb/measures/update_time.hpp

Add files via upload

Update parameters.cpp

Update parameters.hpp

Update solver_core_desc.py

Update impurity_trace.cpp

Update impurity_trace.cpp

Update impurity_trace.cpp

Update impurity_trace.cpp

test

test2

Wentzell commented:
Thank you @ec147 for this Pull Request!

Please add a description of the Feature and the changes to this PR.

Also, can you please adjust the git history to group the changes into
logical chunks with useful commit messages, instead of generic
messages like [Add files via upload](https://github.com/TRIQS/cthyb/pull/178/commits/f884fa4c490e2074bdcaedc35ef091f914e6dd78)

ec147 commented:
So far, my PR cover all my original modifications except bullets 2 and 6. Since the last three bullets are only one-line modifications, I have combined them with the first bullet, but I can do a separate PR for them if you prefer. Likewise, since bullet 6 is only one line and is related to a part of the code that is also modified by bullet 2, I think it's easier to merge them into a single PR, but it's up to you. I haven't done a PR for bullet 2 yet since adding it to the Python API is a bit tricky, so please let me a bit of additional time. And once we agree on the way these new features will be included and called in TRIQS/CTHYB, I will start working on the documentation and testing.

I have also squashed all my commits. Hope it's more to your liking.

Also, I seem to have an issue with your automatic tests. The same two tests keep failing for all of my branches, although they're perfectly fine on my machine. I have done some very basic tests, such as doing a trivial commit (eg adding a blank) on a branch that had previously passed all the tests a few days ago, and now these two tests fail. Do you have any idea what could be causing this ?

ec147 commented:
Hi,
Just to check how the review process is progressing. Would it be possible to have an estimate of how long it is going to take ?

Wentzell commented:
Dear @ec147,

We have started our work on Monte Carlo related improvements with @Thoemi09,
c.f. https://github.com/TRIQS/triqs/pull/962
Once those `mc_generic` changes are finalized we will move to
auto-correlation and error analysis before moving to more specific cthyb topics
including the one in this PR.

I hope this helps, sorry to not have a precise time-line for you.

Wentzell commented:
Thank you again for your work and proposed changes!

ec147 commented:
Dear @Wentzell,

Thanks for the update. I understand, you have some constraints and some deadlines to meet.
But would it be possible to have at least a rough estimate ? Are we talking a few weeks ? A few months ? 
I only have a temporary position in my lab, so I really need to have an idea of how long this is going to take in order to get organized.

Best

Wentzell commented:
Dear @ec147,

We will move to the TRIQS CTQMC improvements in the coming months.

ec147 commented:
I am considering closing all my pull requests for inactivity, clear will on your part of slowing down research, and overall huge lack of respect. Should I ?

ec147 commented:
I have no intention to beg in order to be allowed to reduce your computation and improve your code.

Commit:
Minor fixes on time invariance

Commit:
Add test for time invariance

Commit:
Fixes to time invariance

Commit:
Update test on time invariance

Commit:
Fix zero division issues with sanitizer
